### ISSUE TITLE
流量问题

### ISSUE THREAD

**Issue by cheehong1030** [hktrpg/TG.line.Discord.Roll.Bot#705]
**流量问题**

在这个项目没有考虑分布式和微服务吗 如果流量大了你应该怎么处理 并且编写那么庞大的系统也没有代码规范 不怕以后难以维护吗

**Comment by cheehong1030** [comment]
我也有发现你的discord bot在30000个服务器 那么你现在的服务器配置是什么
还有你服务器怎么大呢 当到了一定规模一定会卡的 因为你不是使用高并发之类的
你可以尝试考虑移去微服务 一个主服务器处理指令 bot接收了指令返回给服务器处理 也许可以考虑这个提议

**Comment by zeteticl** [comment]
@cheehong1030 有推薦的文件看嗎XD

**Comment by cheehong1030** [comment]
没有 但是我可以给你一些微服务例子

主服务器用于处理数据接收数据和处理数据库 你可以使用 REST API/grpc/websocket 用于开放端口
主服务器呢 如果流量大你可以考虑负载均衡 但是如果你有高并发的相关经验也是适用的

在例如Discord/Whatsapp/Telegram这种bot呢 接收指令然后根据主服务器的数据类型返回数据给主服务器处理
这个方法维护和扩展性高

**Comment by zeteticl** [comment]
謝謝, 好像沒什麼文件說太多, 要仔細研究這課題
現在運行起來蠻順,幾次改版後, 也沒有以前時常crash 的情況(以前是一個月至少一次...)

**Comment by zeteticl** [comment]
不過如果分離後, 可以熱升級代碼, 或是可以更新重啟也不錯

**Comment by cheehong1030** [comment]
对 你还可以考虑尝试 redis+数据库组合，这个是为了减少io操作 尝试更多的优化方法而不是一直升级服务器配置 在这个架构你也可以使用热升级技术来确保无缝衔接升级服务器

**Comment by zeteticl** [comment]
> 对 在这个架构可以确保不会让太多服务一次性崩溃 然后你还可以考虑尝试 redis+数据库组合，这个是为了减少io操作 尝试更多的优化方法而不是一直升级服务器配置 你也可以使用热升级技术来确保无缝衔接升级服务器

相隔年半，現在有AI，感覺有機會可以重構了XD
科技世界真的日新月異

**Comment by cheehong1030** [comment]
> > 对 在这个架构可以确保不会让太多服务一次性崩溃 然后你还可以考虑尝试 redis+数据库组合，这个是为了减少io操作 尝试更多的优化方法而不是一直升级服务器配置 你也可以使用热升级技术来确保无缝衔接升级服务器
> 
> 相隔年半，現在有AI，感覺有機會可以重構了XD 科技世界真的日新月異

加油 但是AI重构调试有点麻烦 还得人工介入 还有如果你使用微服务可以尝试docker

**Comment by zeteticl** [comment]
> > > 对 在这个架构可以确保不会让太多服务一次性崩溃 然后你还可以考虑尝试 redis+数据库组合，这个是为了减少io操作 尝试更多的优化方法而不是一直升级服务器配置 你也可以使用热升级技术来确保无缝衔接升级服务器
> > 
> > 
> > 相隔年半，現在有AI，感覺有機會可以重構了XD 科技世界真的日新月異
> 
> 加油 但是AI重构调试有点麻烦 还得人工介入 还有如果你使用微服务可以尝试docker

深有體會, 試用了很多AI, 最近在用cursor, 有時還是會人工智障地把部份CODE 刪了, 大小寫不合 等等問題
不看著會有極可怕後果XDD

**Comment by zeteticl** [comment]
感謝建議，最後用了cursor，把整個服務轉成docker和分散式
不用因為更新了部份代碼，就重啓DISCORD BOT，重啓一次太久了

**Comment by cheehong1030** [comment]
哈哈哈 时隔4年的issue 居然还记得
恭喜你 但是请谨慎使用whatsapp-web.js
